Catch SIGTERM on default settings.

weezel opened the pull request:
This is useful with services like AWS Fargate where SIGTERM is sent when container is being stopped [1]. Can drastically speed up the shutdown and most importantly does it gracefully.

Caveats: won't work on Windows.

[1] https://aws.amazon.com/blogs/containers/graceful-shutdowns-with-ecs/

heppu reviewed (changes requested):
This is a good addition! Before merging this:

1. change the comment line to reflect new defaults
2. add test case for default option sigterm usage

For the test case, you could modify the [TestListener](https://github.com/go-srvc/mods/blob/main/sigmod/signal_test.go#L13) by adding `inputSig syscal.Signal` field and then use that.

weezel commented:
> This is a good addition! Before merging this:
> 
>     1. change the comment line to reflect new defaults
> 
>     2. add test case for default option sigterm usage
> 
> 
> For the test case, you could modify the [TestListener](https://github.com/go-srvc/mods/blob/main/sigmod/signal_test.go#L13) by adding `inputSig syscal.Signal` field and then use that.

Thanks for the input. Now with the tests included and comment modified.

heppu commented on sigmod/signal.go:
```diff
@@ -14,10 +15,13 @@ type Listener struct {
-		signals = []os.Signal{os.Interrupt}
+		signals = []os.Signal{
+			os.Interrupt,
+			syscall.SIGTERM, // XXX Not available on Windows
```
I completely forgot this windows aspect. I will test this on windows and come back with comments.

weezel replied:
I suppose this will continue to work on Windows even there's no mapping for it.

heppu replied:
Idea was to decouple test from that variable, hence the comment it had. More precisely the point was to detect possible breaking changes in behavior in case someone modifies that slice. Now if same slice is used in test we won't detect those.

heppu commented:
Finally got windows machine to implement working win support #25 and now the code uses proper build [constraints](https://pkg.go.dev/cmd/go#hdr-Build_constraints) to achieve win/unix compatibility.

If you rebase the pr you should be able to add sigterm only for unix platfors [here ](https://github.com/go-srvc/mods/blob/4a0eb9dd18a8b342ccad563d4bc9700c035231dd/sigmod/defaults.go#L7)and [here ](https://github.com/go-srvc/mods/blob/4a0eb9dd18a8b342ccad563d4bc9700c035231dd/sigmod/signal_unix_test.go#L16).

weezel committed:
Include SIGTERM into defaults on *nix systems.

On non-Windows systems SIGTERM is included into default signals.

This is useful with services like AWS Fargate where SIGTERM is sent when
container is being stopped [1]. Can drastically speed up the shutdown
and most importantly does it gracefully.

[1] https://aws.amazon.com/blogs/containers/graceful-shutdowns-with-ecs/

Signed-off-by: Ville Valkonen <weezel@users.noreply.github.com>

weezel commented on sigmod/defaults_win.go:
```diff
@@ -4,4 +4,4 @@ package sigmod
 import "os"
 
-var defaultSignals = []os.Signal{os.Kill}
+var DefaultSignals = []os.Signal{os.Kill}
```
Needed to export this value to make it available for tests or then I overlooked something. Don't really see any harm if it's exported, though.

heppu replied:
Idea was to decouple test from that variable, hence the comment it had on test file. More precisely the point is to detect possible breaking changes in behavior in case someone modifies the default values. Now if same slice is used in test we won't detect those.

weezel replied:
Reread what you wrote earlier and overlooked the crucial parts, sorry for that. Hopefully the solution provided now is more fullbrained than the previous one :upside_down_face:
